--- origsrc/google-gadgets-for-linux-0.10.5/ggadget/basic_element.cc	2008-12-11 04:08:24.000000000 -0600
+++ src/google-gadgets-for-linux-0.10.5/ggadget/basic_element.cc	2009-01-29 11:09:11.799238500 -0600
@@ -1008,7 +1008,7 @@ class BasicElement::Impl : public SmallO
   ImageInterface *mask_image_;
   CanvasInterface *cache_;
   const char *tag_name_;
-  size_t index_;
+  int32_t index_;
 
   double width_, height_, pwidth_, pheight_;
   double x_, y_, px_, py_;
@@ -1649,11 +1649,11 @@ void BasicElement::SetParentElement(Basi
   impl_->parent_ = parent;
 }
 
-size_t BasicElement::GetIndex() const {
+int32_t BasicElement::GetIndex() const {
   return impl_->index_;
 }
 
-void BasicElement::SetIndex(size_t index) {
+void BasicElement::SetIndex(int32_t index) {
   impl_->index_ = index;
 }
 
@@ -1986,7 +1986,7 @@ Variant BasicElement::GetPixelOrRelative
   } else {
     // FIXME: Is it necessary to do round here?
     // Answer: to be compatible with Windows version.
-    return Variant(static_cast<int>(round(pixel)));
+    return Variant(static_cast<int32_t>(round(pixel)));
   }
 }
 
--- origsrc/google-gadgets-for-linux-0.10.5/ggadget/basic_element.h	2008-12-23 23:58:52.000000000 -0600
+++ src/google-gadgets-for-linux-0.10.5/ggadget/basic_element.h	2009-01-29 11:09:47.377363500 -0600
@@ -119,8 +119,8 @@ class BasicElement: public ScriptableHel
    * with the element tree structure.
    * Only for @c Elements class which knows how to keep integrity.
    */
-  size_t GetIndex() const;
-  void SetIndex(size_t index);
+  int32_t GetIndex() const;
+  void SetIndex(int32_t index);
   //@}
 
   /**
