--- origsrc/google-gadgets-for-linux-0.10.5/hosts/gtk/Makefile.am	2009-01-08 01:00:32.000000000 -0600
+++ src/google-gadgets-for-linux-0.10.5/hosts/gtk/Makefile.am	2009-01-29 14:45:20.346113500 -0600
@@ -101,7 +101,7 @@ noinst_SCRIPTS=ggl-gtk.sh
 
 .PHONY: ggl-gtk.sh
 
-ggl-gtk.sh: ggl-gtk
+ggl-gtk.sh: ggl-gtk$(EXEEXT)
 	(echo '#!/bin/sh' > $@; \
 	 echo 'GGL_MODULE_PATH=`pwd`' >> $@; \
 	 echo 'for i in $(abs_top_builddir)/extensions/*; do' >> $@; \
--- origsrc/google-gadgets-for-linux-0.10.5/hosts/qt/Makefile.am	2009-01-08 01:00:57.000000000 -0600
+++ src/google-gadgets-for-linux-0.10.5/hosts/qt/Makefile.am	2009-01-29 13:33:26.283613500 -0600
@@ -100,7 +100,7 @@ noinst_SCRIPTS=ggl-qt.sh
 
 .PHONY: ggl-qt.sh
 
-ggl-qt.sh: ggl-qt
+ggl-qt.sh: ggl-qt$(EXEEXT)
 	(echo '#!/bin/sh' > $@; \
 	 echo 'GGL_MODULE_PATH=`pwd`' >> $@; \
 	 echo 'for i in $(abs_top_builddir)/extensions/*; do' >> $@; \
--- origsrc/google-gadgets-for-linux-0.10.5/xdg/Makefile.am	2009-01-08 01:04:08.000000000 -0600
+++ src/google-gadgets-for-linux-0.10.5/xdg/Makefile.am	2009-01-29 13:11:46.799238500 -0600
@@ -34,13 +34,13 @@ noinst_SCRIPTS=intl-desktop-file.sh
 .PHONY: intl-desktop-file.sh
 
 if GGL_CROSS_COMPILING
-intl-desktop-file.sh: intl-desktop-file
+intl-desktop-file.sh: intl-desktop-file$(EXEEXT)
 	@echo '#!/bin/sh' > $@; \
 	 echo 'echo intl-desktop-file can not work in cross-compiling mode.' >> $@; \
 	 echo 'cp $$@' >> $@; \
 	 chmod 0755 $@
 else
-intl-desktop-file.sh: intl-desktop-file
+intl-desktop-file.sh: intl-desktop-file$(EXEEXT)
 	@echo '#!/bin/sh' > $@; \
 	 echo 'GGL_MODULE_PATH=`pwd`' >> $@; \
 	 echo 'for i in $(abs_top_builddir)/extensions/*; do' >> $@; \
