--- origsrc/google-gadgets-for-linux-0.11.0/extensions/analytics_usage_collector/Makefile.am	2009-03-14 23:12:42.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/analytics_usage_collector/Makefile.am	2009-06-23 17:18:01.853957200 -0500
@@ -33,5 +33,6 @@ analytics_usage_collector_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 analytics_usage_collector_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/curl_xml_http_request/Makefile.am	2009-03-14 23:12:43.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/curl_xml_http_request/Makefile.am	2009-06-23 17:18:01.857707300 -0500
@@ -38,6 +38,7 @@ curl_xml_http_request_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 curl_xml_http_request_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/dbus_script_class/Makefile.am	2009-03-14 23:12:44.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/dbus_script_class/Makefile.am	2009-06-23 17:18:01.861457300 -0500
@@ -42,6 +42,7 @@ dbus_script_class_la_LIBADD = \
 			  $(top_builddir)/ggadget/dbus/libggadget-dbus@GGL_EPOCH@.la
 
 dbus_script_class_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 endif
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/default_framework/Makefile.am	2009-03-14 23:12:43.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/default_framework/Makefile.am	2009-06-23 17:18:01.863957300 -0500
@@ -33,5 +33,6 @@ default_framework_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 default_framework_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/default_options/Makefile.am	2009-03-14 23:12:43.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/default_options/Makefile.am	2009-06-23 17:18:01.867707400 -0500
@@ -37,6 +37,7 @@ default_options_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 default_options_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/google_gadget_manager/Makefile.am	2009-03-14 23:12:42.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/google_gadget_manager/Makefile.am	2009-06-23 17:18:01.868957400 -0500
@@ -41,5 +41,6 @@ google_gadget_manager_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 google_gadget_manager_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/gst_audio_framework/Makefile.am	2009-03-14 23:12:43.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/gst_audio_framework/Makefile.am	2009-06-23 17:18:01.871457500 -0500
@@ -37,6 +37,7 @@ gst_audio_framework_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 gst_audio_framework_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/gst_video_element/Makefile.am	2009-03-14 23:12:43.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/gst_video_element/Makefile.am	2009-06-23 17:18:01.873957500 -0500
@@ -40,6 +40,7 @@ gst_video_element_la_LIBADD = $(GSTREAME
 			      $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 gst_video_element_la_LDFLAGS = -avoid-version \
+			       -no-undefined \
 			       -module
 
 endif
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/gtk_edit_element/Makefile.am	2009-03-14 23:12:43.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/gtk_edit_element/Makefile.am	2009-06-23 17:18:01.876457500 -0500
@@ -43,6 +43,7 @@ gtk_edit_element_la_LIBADD = \
 			  $(top_builddir)/ggadget/gtk/libggadget-gtk@GGL_EPOCH@.la
 
 gtk_edit_element_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 endif
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/gtk_flash_element/Makefile.am	2009-03-14 23:12:45.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/gtk_flash_element/Makefile.am	2009-06-23 17:18:01.877707600 -0500
@@ -34,6 +34,7 @@ gtk_flash_element_la_LIBADD = $(top_buil
 			      $(top_builddir)/ggadget/gtk/libggadget-gtk@GGL_EPOCH@.la
 
 gtk_flash_element_la_LDFLAGS = -avoid-version \
+			       -no-undefined \
 			       -module
 
 endif
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/gtk_system_framework/Makefile.am	2009-03-14 23:12:44.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/gtk_system_framework/Makefile.am	2009-06-23 17:18:01.880207600 -0500
@@ -39,6 +39,7 @@ gtk_system_framework_la_LIBADD = \
 			  $(top_builddir)/ggadget/xdg/libggadget-xdg@GGL_EPOCH@.la
 
 gtk_system_framework_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/gtkmoz_browser_element/Makefile.am	2009-03-14 23:12:43.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/gtkmoz_browser_element/Makefile.am	2009-06-23 17:18:01.882707600 -0500
@@ -42,6 +42,7 @@ gtkmoz_browser_element_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 gtkmoz_browser_element_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/gtkwebkit_browser_element/Makefile.am	2009-05-10 22:30:34.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/gtkwebkit_browser_element/Makefile.am	2009-06-23 17:19:02.203629300 -0500
@@ -41,6 +41,7 @@ gtkwebkit_browser_element_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 gtkwebkit_browser_element_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/html_flash_element/Makefile.am	2009-05-10 22:30:34.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/html_flash_element/Makefile.am	2009-06-23 17:19:20.628910900 -0500
@@ -29,4 +29,4 @@ html_flash_element_la_CPPFLAGS = $(PREDE
 
 html_flash_element_la_LIBADD = $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
-html_flash_element_la_LDFLAGS = -avoid-version -module
+html_flash_element_la_LDFLAGS = -no-undefined -avoid-version -module
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/libxml2_xml_parser/Makefile.am	2009-03-14 23:12:44.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/libxml2_xml_parser/Makefile.am	2009-06-23 17:18:01.885207700 -0500
@@ -37,6 +37,7 @@ libxml2_xml_parser_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 libxml2_xml_parser_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/linux_system_framework/Makefile.am	2009-03-14 23:12:42.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/linux_system_framework/Makefile.am	2009-06-24 00:48:39.462242700 -0500
@@ -103,6 +103,7 @@ linux_system_framework_la_LIBADD += \
 endif
 
 linux_system_framework_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/qt_edit_element/Makefile.am	2009-03-14 23:12:43.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/qt_edit_element/Makefile.am	2009-06-23 17:18:01.887707700 -0500
@@ -41,6 +41,7 @@ qt_edit_element_la_LIBADD = \
 			  $(top_builddir)/ggadget/qt/libggadget-qt@GGL_EPOCH@.la
 
 qt_edit_element_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 endif
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/qt_script_runtime/Makefile.am	2009-03-14 23:12:45.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/qt_script_runtime/Makefile.am	2009-06-23 17:18:01.888957700 -0500
@@ -71,6 +71,7 @@ qt_script_runtime_la_LIBADD = \
 			  libggadget-qtjs.la
 
 qt_script_runtime_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/qt_system_framework/Makefile.am	2009-03-14 23:12:44.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/qt_system_framework/Makefile.am	2009-06-23 17:18:01.891457800 -0500
@@ -38,6 +38,7 @@ qt_system_framework_la_LIBADD = \
 			  $(top_builddir)/ggadget/xdg/libggadget-xdg@GGL_EPOCH@.la
 
 qt_system_framework_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/qt_xml_http_request/Makefile.am	2009-03-14 23:12:45.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/qt_xml_http_request/Makefile.am	2009-06-23 17:18:01.893957800 -0500
@@ -44,6 +44,7 @@ qt_xml_http_request_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 qt_xml_http_request_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/qtwebkit_browser_element/Makefile.am	2009-03-14 23:12:44.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/qtwebkit_browser_element/Makefile.am	2009-06-23 17:18:01.896457800 -0500
@@ -45,6 +45,7 @@ qtwebkit_browser_element_la_LIBADD = \
 			  $(top_builddir)/ggadget/qt/libggadget-qt@GGL_EPOCH@.la
 
 qtwebkit_browser_element_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 BUILT_SOURCES		= $(MOC_FILES)
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/smjs_script_runtime/Makefile.am	2009-03-14 23:12:44.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/smjs_script_runtime/Makefile.am	2009-06-23 17:18:01.898957900 -0500
@@ -77,6 +77,7 @@ smjs_script_runtime_la_LIBADD = \
 
 smjs_script_runtime_la_LDFLAGS = \
 			  $(LIBMOZJS_LDFLAGS) \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 endif
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/soup_xml_http_request/Makefile.am	2009-05-19 01:05:10.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/soup_xml_http_request/Makefile.am	2009-06-23 17:18:27.746852800 -0500
@@ -39,6 +39,7 @@ soup_xml_http_request_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 soup_xml_http_request_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/extensions/webkit_script_runtime/Makefile.am	2009-05-10 22:30:34.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/extensions/webkit_script_runtime/Makefile.am	2009-06-23 17:19:55.283190400 -0500
@@ -48,6 +48,11 @@ libggadget_webkitjs@GGL_EPOCH@_la_LIBADD
 				$(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la \
 				$(top_builddir)/ggadget/js/libggadget-js@GGL_EPOCH@.la
 
+libggadget_webkitjs@GGL_EPOCH@_la_LDFLAGS = \
+				-version-info $(LIBGGADGET_VERSION) \
+				-no-undefined \
+				-export-dynamic
+
 extension_LTLIBRARIES = webkit-script-runtime.la
 extensiondir = $(GGL_MODULE_DIR)
 
@@ -67,6 +72,7 @@ webkit_script_runtime_la_LIBADD = \
 				libggadget-webkitjs@GGL_EPOCH@.la
 
 webkit_script_runtime_la_LDFLAGS = \
+				-no-undefined \
 				-avoid-version \
 				-module
 
--- origsrc/google-gadgets-for-linux-0.11.0/gadgets/designer/Makefile.am	2009-03-14 23:12:54.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/gadgets/designer/Makefile.am	2009-06-23 17:18:01.901457900 -0500
@@ -37,6 +37,7 @@ designer_script_utils_la_LIBADD = \
 			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 
 designer_script_utils_la_LDFLAGS = \
+			  -no-undefined \
 			  -avoid-version \
 			  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/ggadget/Makefile.am	2009-03-20 04:30:59.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/ggadget/Makefile.am	2009-06-23 17:18:01.903958000 -0500
@@ -257,6 +257,7 @@ libggadget@GGL_EPOCH@_la_LIBADD = \
 
 libggadget@GGL_EPOCH@_la_LDFLAGS = \
 			  -version-info $(LIBGGADGET_VERSION) \
+			  -no-undefined \
 			  -export-dynamic \
 			  $(LIBLTDL)
 
--- origsrc/google-gadgets-for-linux-0.11.0/ggadget/dbus/Makefile.am	2009-03-14 23:12:49.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/ggadget/dbus/Makefile.am	2009-06-23 17:18:01.906458000 -0500
@@ -48,6 +48,7 @@ libggadget_dbus@GGL_EPOCH@_la_LIBADD = \
 
 libggadget_dbus@GGL_EPOCH@_la_LDFLAGS = \
 			  -version-info $(LIBGGADGET_VERSION) \
+			  -no-undefined \
 			  -export-dynamic
 
 pkgconfigdir= $(libdir)/pkgconfig
--- origsrc/google-gadgets-for-linux-0.11.0/ggadget/gtk/Makefile.am	2009-03-14 23:12:51.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/ggadget/gtk/Makefile.am	2009-06-23 17:18:01.908958000 -0500
@@ -72,6 +72,7 @@ libggadget_gtk@GGL_EPOCH@_la_LIBADD = \
 
 libggadget_gtk@GGL_EPOCH@_la_LDFLAGS = \
 			  -version-info $(LIBGGADGET_VERSION) \
+			  -no-undefined \
 			  -export-dynamic
 
 if GTK2_TARGET_X11
--- origsrc/google-gadgets-for-linux-0.11.0/ggadget/js/Makefile.am	2009-03-14 23:12:48.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/ggadget/js/Makefile.am	2009-06-23 17:18:01.911458100 -0500
@@ -43,6 +43,7 @@ libggadget_js@GGL_EPOCH@_la_LIBADD = \
 
 libggadget_js@GGL_EPOCH@_la_LDFLAGS = \
 			  -version-info $(LIBGGADGET_VERSION) \
+			  -no-undefined \
 			  -export-dynamic
 
 jscript_massager.cc : jscript_massager.l
--- origsrc/google-gadgets-for-linux-0.11.0/ggadget/npapi/Makefile.am	2009-03-14 23:12:50.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/ggadget/npapi/Makefile.am	2009-06-23 17:18:01.912708100 -0500
@@ -51,6 +51,7 @@ libggadget_npapi@GGL_EPOCH@_la_LIBADD = 
 
 libggadget_npapi@GGL_EPOCH@_la_LDFLAGS = \
 			  -version-info $(LIBGGADGET_VERSION) \
+			  -no-undefined \
 			  -export-dynamic
 
 pkgconfigdir= $(libdir)/pkgconfig
--- origsrc/google-gadgets-for-linux-0.11.0/ggadget/qt/Makefile.am	2009-03-14 23:12:46.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/ggadget/qt/Makefile.am	2009-06-23 17:18:01.915208100 -0500
@@ -75,6 +75,7 @@ libggadget_qt@GGL_EPOCH@_la_LIBADD = \
 
 libggadget_qt@GGL_EPOCH@_la_LDFLAGS = \
 			  -version-info $(LIBGGADGET_VERSION) \
+			  -no-undefined \
 			  -export-dynamic
 
 pkgconfigdir= $(libdir)/pkgconfig
--- origsrc/google-gadgets-for-linux-0.11.0/ggadget/tests/Makefile.am	2009-03-14 23:12:50.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/ggadget/tests/Makefile.am	2009-06-23 17:18:01.917708200 -0500
@@ -116,6 +116,7 @@ foo_module_la_CPPFLAGS		= $(PREDEFINED_M
 foo_module_la_LIBADD		= $(top_builddir)/unittest/libgtest.la \
 				  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 foo_module_la_LDFLAGS		= -avoid-version \
+				  -no-undefined \
 				  -rpath ${abs_builddir} \
 				  -module
 
@@ -127,6 +128,7 @@ bar_module_la_CPPFLAGS		= $(PREDEFINED_M
 bar_module_la_LIBADD		= $(top_builddir)/unittest/libgtest.la \
 				  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la
 bar_module_la_LDFLAGS		= -avoid-version \
+				  -no-undefined \
 				  -rpath ${abs_builddir} \
 				  -module
 
--- origsrc/google-gadgets-for-linux-0.11.0/ggadget/xdg/Makefile.am	2009-03-14 23:12:52.000000000 -0500
+++ src/google-gadgets-for-linux-0.11.0/ggadget/xdg/Makefile.am	2009-06-23 17:18:01.920208200 -0500
@@ -50,6 +50,7 @@ endif
 
 libggadget_xdg@GGL_EPOCH@_la_LDFLAGS = \
 			  -version-info $(LIBGGADGET_VERSION) \
+			  -no-undefined \
 			  -export-dynamic
 
 pkgconfigdir = $(libdir)/pkgconfig
